src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.diff
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.services.HmppsDomainEventEmitter
+import uk.gov.justice.digital.hmpps.prisonersearch.services.HmppsDomainEventEmitter.PrisonerReceiveReason.TRANSFERRED
+import uk.gov.justice.digital.hmpps.prisonersearch.services.diff.PossibleMovementChange.MovementChange.TransferIn
+
+@Service
+class PrisonerMovementsEventService(private val domainEventEmitter: HmppsDomainEventEmitter) {
+  fun generateAnyMovementEvents(
+    previousPrisonerSnapshot: Prisoner?,
+    prisoner: Prisoner
```

Comment:
Nice improvement in language - be good to make it consistent across the old events too.